src/app/shared/navigation/navigation.component.ts

```diff
@@ -76,14 +88,39 @@ export class NavigationComponent {
   public constructor(
     private readonly navigationListService: NavigationListService,
     private readonly preferenceService: PreferenceService,
-    private readonly activatedRoute: ActivatedRoute
+    private readonly navListComponentService: NavigationListComponentService,
+    private readonly activatedRoute: ActivatedRoute,
+    private readonly timeRangeService: TimeRangeService
   ) {
-    this.navItems = this.navItemDefinitions.map(definition =>
+    const navItems = this.navItemDefinitions.map(definition =>
       this.navigationListService.decorateNavItem(definition, this.activatedRoute)
     );
+    // Decorate the nav items with the corresponding time ranges, depending on the FF state.
+    // The time ranges in nav items are streams that get the most recent value from page time range preference service
+    this.navItems$ = this.navListComponentService.resolveNavItemConfigTimeRanges(navItems).pipe(shareReplay());
```

Comment:
do we subscribe to this more than once? If there's a reason to share it, it's not obvious so should be commented.

Reply:
nope, removed. it was for a previous implementation i believe